### DISCUSSION TITLE
Debounced input with async resolver

### DISCUSSION THREAD

**Discussion by jodoox** [react-hook-form/react-hook-form#10775] [Q&A]
I've been trying to get a debounced username field work with async validation. So far I've not managed to make it work and I'd appreciate any pointers. Here is a simple example reproducing the issue, the only difference between the two forms being that the async validation callback is debounced (via the `use-debounce/useDebouncedCallback` hook) or not.

https://codesandbox.io/p/sandbox/async-debounced-resolver-8lxtmt

My issue here is that the form validation is late in the debounced version: despite the `isValid` boolean being received, this does not get reflected on the form's state until the next validation (onBlur / onChange).

![CleanShot 2023-08-07 at 13 54 23](https://github.com/react-hook-form/react-hook-form/assets/10785783/9c3b841d-dab2-40d1-a9fe-f6bf5262b303)

Happy to move this to another place (stackoverflow?) if this question does not belong here- the discussions in @hookform/resolvers are closed so I figured I'd ask here.

**Comment by musjj** [comment]
Did you managed to solve this? The problem is that the debounced function always returns the result of the last invocation (or `undefined` in the first invocation). I think `use-debounce` is most suitable for functions that invokes side effects somewhere else.
